src/input/findReportedConfiguration.test.ts

```diff
@@ -56,4 +56,23 @@ describe("findReportedConfiguration", () => {
             rules,
         });
     });
+
+    it("declares a correct DeepPartial type", () => {
```

Comment:
Hmm, I don't think we particularly need to unit test for type system types. I think now that you've fleshed out usage of the `DeepPartial` type, it's apparent through the rest of the code whether and how it works.

What do you think?

Reply:
Fine by me, the test is a little goofy compared to other tests in this project. I just wrote it first and then fixed all the rest of the code by iteratively rebuilding and running tests and making small changes until everything worked out.

I'll drop the test in another commit shortly.